src/aal/aal.h

```diff
@@ -46,6 +46,16 @@ namespace snmalloc
   template<class Arch>
   struct AAL_Generic : Arch
   {
+    struct default_address_t
```

Comment:
Please add some comments here.  It took me a little while to figure out what this code is doing.